### PR TITLE
Style/suppressions

### PR DESCRIPTION
**Pull request by marvin-bitterlich**
fixes a grim I had a long time now

Review please

### CODE REVIEW

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1085/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

File: src/main/java/com/minecolonies/coremod/colony/Structures.java [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import com.minecolonies.api.util.LanguageHandler;
 import com.minecolonies.api.util.Log;
 import com.minecolonies.api.util.constant.Constants;
+import com.minecolonies.api.util.constant.Suppression;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/Structures.java]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.Suppression'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java [hunk 2]
```diff
@@ -4,6 +4,7 @@
 import com.minecolonies.api.util.InventoryUtils;
 import com.minecolonies.api.util.MathUtils;
 import com.minecolonies.api.util.Utils;
+import com.minecolonies.api.util.constant.Suppression;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.Suppression'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java [hunk 3]
```diff
@@ -531,7 +533,7 @@ private int findSaplingSlot()
     }
 
     //todo: we need to use a different way to get Metadata
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java [hunk 4]
```diff
@@ -553,7 +555,7 @@ private void placeSaplings(final int saplingSlot, @NotNull final ItemStack stack
...
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java:556]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 5]
```diff
@@ -230,6 +236,7 @@ public AxisAlignedBB getCollisionBoundingBox(final IBlockState blockState, final
     /**
      * @deprecated (Remove this as soon as minecraft offers anything better).
      */
+    @SuppressWarnings(DEPRECATION)
     @Override
     @Deprecated
     public boolean isOpaqueCube(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java:242]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 6]
```diff
@@ -219,7 +224,8 @@ public boolean shouldSideBeRendered(final IBlockState blockState, final IBlockAc
     /**
      * @deprecated (Remove this as soon as minecraft offers anything better).
      */
-    @javax.annotation.Nullable
+    @SuppressWarnings(DEPRECATION)
+    @Nullable
     @Deprecated
     @Override
     public AxisAlignedBB getCollisionBoundingBox(final IBlockState blockState, final IBlockAccess worldIn, final BlockPos pos)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java:231]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 7]
```diff
@@ -164,6 +166,7 @@ public int getMetaFromState(@NotNull final IBlockState state)
     /**
      * @deprecated (Remove this as soon as minecraft offers anything better).
      */
+    @SuppressWarnings(DEPRECATION)
     @Override
     @Deprecated
     public boolean isFullCube(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java:172]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 8]
```diff
@@ -180,6 +183,7 @@ public boolean isPassable(final IBlockAccess worldIn, final BlockPos pos)
     /**
      * @deprecated (Remove this as soon as minecraft offers anything better).
      */
+    @SuppressWarnings(DEPRECATION)
     @Override
     @Deprecated
     public AxisAlignedBB getBoundingBox(final IBlockState state, final IBlockAccess source, final BlockPos pos)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java:189]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 9]
```diff
@@ -6,6 +6,7 @@
 import com.minecolonies.api.configuration.Configurations;
 import com.minecolonies.api.util.*;
 import com.minecolonies.api.util.constant.Constants;
+import com.minecolonies.api.util.constant.Suppression;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.Suppression'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java [hunk 10]
```diff
@@ -762,7 +764,7 @@ public void resetCurrentStructure()
      * The rule thinks we can merge the two forge loops iterating over resources
      * But in this case the rule does not apply because that would destroy the logic.
      */
-    @SuppressWarnings("squid:S3047")
+    @SuppressWarnings(MULTIPLE_LOOPS_OVER_THE_SAME_SET_SHOULD_BE_COMBINED)
     private Boolean spawnEntity(@NotNull final Structure.StructureBlock currentBlock)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java:767]: MAJOR Refactor this method to reduce its Cognitive Complexity from 32 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java [hunk 11]
```diff
@@ -2,6 +2,7 @@
 
 import com.minecolonies.api.configuration.Configurations;
 import com.minecolonies.api.util.*;
+import com.minecolonies.api.util.constant.Suppression;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.Suppression'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/blocks/BlockSubstitution.java [hunk 12]
```diff
@@ -1,13 +1,16 @@
 package com.minecolonies.coremod.blocks;
 
 import com.minecolonies.api.util.constant.Constants;
+import com.minecolonies.api.util.constant.Suppression;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockSubstitution.java]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.Suppression'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/blocks/BlockSubstitution.java [hunk 13]
```diff
@@ -63,7 +66,7 @@ private void initBlock()
      * @return true
      */
     //todo: remove once we no longer need to support this
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockSubstitution.java:67]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/sounds/DeliverymanSounds.java [hunk 14]
```diff
@@ -1,5 +1,6 @@
 package com.minecolonies.coremod.sounds;
 
+import com.minecolonies.api.util.constant.Suppression;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/sounds/DeliverymanSounds.java]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.Suppression'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/blocks/BlockHutField.java [hunk 15]
```diff
@@ -1,6 +1,7 @@
 package com.minecolonies.coremod.blocks;
 
 import com.minecolonies.api.util.constant.Constants;
+import com.minecolonies.api.util.constant.Suppression;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockHutField.java]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.Suppression'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/blocks/BlockHutField.java [hunk 16]
```diff
@@ -146,7 +148,7 @@ public AxisAlignedBB getBoundingBox(final IBlockState state, final IBlockAccess
     }
 
     //todo: remove once we no longer need to support this
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockHutField.java]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/blocks/BlockHutField.java [hunk 17]
```diff
@@ -133,7 +135,7 @@ public boolean isPassable(final IBlockAccess worldIn, final BlockPos pos)
...
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockHutField.java]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/blocks/BlockHutField.java [hunk 18]
```diff
@@ -118,7 +120,7 @@ public int getMetaFromState(@NotNull final IBlockState state)
     }
 
     //todo: remove once we no longer need to support this
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockHutField.java:121]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java [hunk 19]
```diff
@@ -172,7 +174,7 @@ public void requestMaterialsIfRequired()
      * The rule thinks we should have less continue and breaks.
      * But in this case the rule does not apply because code would become unreadable and uneffective without.
      */
-    @SuppressWarnings("squid:S135")
+    @SuppressWarnings(LOOPS_SHOULD_NOT_CONTAIN_MORE_THAN_A_SINGLE_BREAK_OR_CONTINUE_STATEMENT)
     private void requestMaterials()
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java:177]: MAJOR The Cyclomatic Complexity of this method "requestMaterials" is 17 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java [hunk 20]
```diff
@@ -2,6 +2,7 @@
 
 import com.minecolonies.api.configuration.Configurations;
 import com.minecolonies.api.util.*;
+import com.minecolonies.api.util.constant.Suppression;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.Suppression'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java [hunk 21]
```diff
@@ -1,6 +1,7 @@
 package com.minecolonies.coremod.blocks;
 
 import com.minecolonies.api.util.constant.Constants;
+import com.minecolonies.api.util.constant.Suppression;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.Suppression'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/colony/jobs/AbstractJob.java [hunk 22]
```diff
@@ -1,6 +1,7 @@
 package com.minecolonies.coremod.colony.jobs;
 
 import com.minecolonies.api.util.Log;
+import com.minecolonies.api.util.constant.Suppression;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/jobs/AbstractJob.java]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.Suppression'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/blocks/AbstractBlockHut.java [hunk 23]
```diff
@@ -91,6 +94,7 @@ public TileEntity createNewTileEntity(final World world, final int meta)
     /**
      * @deprecated (Remove this as soon as minecraft offers anything better).
      */
+    @SuppressWarnings(DEPRECATION)
     @Override
     @Deprecated
     public boolean isFullBlock(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/AbstractBlockHut.java:99]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/AbstractBlockHut.java [hunk 24]
```diff
@@ -3,6 +3,7 @@
 import com.minecolonies.api.colony.permissions.Action;
 import com.minecolonies.api.entity.ai.citizen.builder.IBuilderUndestroyable;
 import com.minecolonies.api.util.constant.Constants;
+import com.minecolonies.api.util.constant.Suppression;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/AbstractBlockHut.java]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.Suppression'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/blocks/AbstractBlockHut.java [hunk 25]
```diff
@@ -168,6 +176,7 @@ public boolean isFullCube(final IBlockState state)
     /**
      * @deprecated (Remove this as soon as minecraft offers anything better).
      */
+    @SuppressWarnings(DEPRECATION)
     @Override
     @Deprecated
     public boolean isOpaqueCube(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/AbstractBlockHut.java:181]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/AbstractBlockHut.java [hunk 26]
```diff
@@ -158,6 +165,7 @@ public IBlockState withMirror(@NotNull final IBlockState state, final Mirror mir
     /**
      * @deprecated (Remove this as soon as minecraft offers anything better).
      */
+    @SuppressWarnings(DEPRECATION)
     @Override
     @Deprecated
     public boolean isFullCube(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/AbstractBlockHut.java:170]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockBarrel.java [hunk 27]
```diff
@@ -155,7 +158,7 @@ public int getMetaFromState(final IBlockState state)
      * @return true
      */
     //todo: remove once we no longer need to support this
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockBarrel.java:159]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/blocks/BlockBarrel.java [hunk 28]
```diff
@@ -2,6 +2,7 @@
 
 import com.minecolonies.api.util.Log;
 import com.minecolonies.api.util.constant.Constants;
+import com.minecolonies.api.util.constant.Suppression;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockBarrel.java]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.Suppression'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/blocks/BlockBarrel.java [hunk 29]
```diff
@@ -135,7 +138,7 @@ public void GetItemFromBarrel(final World worldIn, final EntityPlayer playerIn,
     }
 
     //todo: remove once we no longer need to support this
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockBarrel.java:139]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/colony/ColonyList.java [hunk 30]
```diff
@@ -14,6 +15,8 @@
 import java.util.stream.Stream;
 import java.util.stream.StreamSupport;
 
+import static com.minecolonies.api.util.constant.Suppression.*;
+
 //TODO extend list/collection
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/ColonyList.java]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/colony/ColonyList.java [hunk 31]
```diff
@@ -2,6 +2,7 @@
 
 import com.google.common.annotations.VisibleForTesting;
 import com.minecolonies.api.util.Log;
+import com.minecolonies.api.util.constant.Suppression;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/ColonyList.java]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.Suppression'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/colony/workorders/WorkOrderBuildDecoration.java [hunk 32]
```diff
@@ -3,6 +3,7 @@
 import com.minecolonies.api.util.BlockPosUtil;
 import com.minecolonies.api.util.LanguageHandler;
 import com.minecolonies.api.util.Log;
+import com.minecolonies.api.util.constant.Suppression;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/workorders/WorkOrderBuildDecoration.java]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.Suppression'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/colony/workorders/AbstractWorkOrder.java [hunk 33]
```diff
@@ -1,6 +1,7 @@
 package com.minecolonies.coremod.colony.workorders;
 
 import com.minecolonies.api.util.Log;
+import com.minecolonies.api.util.constant.Suppression;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/workorders/AbstractWorkOrder.java]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.Suppression'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/structures/helpers/Structure.java [hunk 34]
```diff
@@ -4,6 +4,7 @@
 import com.minecolonies.api.util.BlockUtils;
 import com.minecolonies.api.util.Log;
 import com.minecolonies.api.util.constant.Constants;
+import com.minecolonies.api.util.constant.Suppression;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/structures/helpers/Structure.java]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.Suppression'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/blocks/BlockSolidSubstitution.java [hunk 35]
```diff
@@ -1,13 +1,16 @@
 package com.minecolonies.coremod.blocks;
 
 import com.minecolonies.api.util.constant.Constants;
+import com.minecolonies.api.util.constant.Suppression;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockSolidSubstitution.java]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.Suppression'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/blocks/BlockSolidSubstitution.java [hunk 36]
```diff
@@ -63,7 +66,7 @@ private void initBlock()
      * @return true
      */
     //todo: remove once we no longer need to support this
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockSolidSubstitution.java:67]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java [hunk 37]
```diff
@@ -553,7 +554,7 @@ private void placeSaplings(final int saplingSlot, @NotNull final ItemStack stack
...
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java:556]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/colony/ColonyList.java [hunk 38]
```diff
@@ -14,6 +14,8 @@
 import java.util.stream.Stream;
 import java.util.stream.StreamSupport;
 
+import static com.minecolonies.api.util.constant.Suppression.UNCHECKED;
+
 //TODO extend list/collection
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/ColonyList.java:19]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/blocks/BlockHutField.java [hunk 39]
```diff
@@ -146,7 +147,7 @@ public AxisAlignedBB getBoundingBox(final IBlockState state, final IBlockAccess
...
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockHutField.java:149]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/blocks/BlockHutField.java [hunk 40]
```diff
@@ -133,7 +134,7 @@ public boolean isPassable(final IBlockAccess worldIn, final BlockPos pos)
...
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockHutField.java:135]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 105 issues
* ![MAJOR][MAJOR] 40 major
* ![MINOR][MINOR] 30 minor
* ![INFO][INFO] 35 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Structures.java#L180](https://github.com/Minecolonies/minecolonies/blob/8a1c3c4a2518b447631ad614a066acafbb44cb24/src/main/java/com/minecolonies/coremod/colony/Structures.java#L180): Either log or rethrow this exception. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [Structures.java#L529](https://github.com/Minecolonies/minecolonies/blob/8a1c3c4a2518b447631ad614a066acafbb44cb24/src/main/java/com/minecolonies/coremod/colony/Structures.java#L529): Reduce this class from 118 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [AbstractJob.java#L61](https://github.com/Minecolonies/minecolonies/blob/8a1c3c4a2518b447631ad614a066acafbb44cb24/src/main/java/com/minecolonies/coremod/colony/jobs/AbstractJob.java#L61): Remove this reference to "JobPlaceholder". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
1. ![MAJOR][MAJOR] [AbstractJob.java#L62](https://github.com/Minecolonies/minecolonies/blob/8a1c3c4a2518b447631ad614a066acafbb44cb24/src/main/java/com/minecolonies/coremod/colony/jobs/AbstractJob.java#L62): Remove this reference to "JobBuilder". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
1. ![MAJOR][MAJOR] [AbstractJob.java#L63](https://github.com/Minecolonies/minecolonies/blob/8a1c3c4a2518b447631ad614a066acafbb44cb24/src/main/java/com/minecolonies/coremod/colony/jobs/AbstractJob.java#L63): Remove this reference to "JobDeliveryman". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
1. ![MAJOR][MAJOR] [AbstractJob.java#L64](https://github.com/Minecolonies/minecolonies/blob/8a1c3c4a2518b447631ad614a066acafbb44cb24/src/main/java/com/minecolonies/coremod/colony/jobs/AbstractJob.java#L64): Remove this reference to "JobMiner". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
1. ![MAJOR][MAJOR] [AbstractJob.java#L65](https://github.com/Minecolonies/minecolonies/blob/8a1c3c4a2518b447631ad614a066acafbb44cb24/src/main/java/com/minecolonies/coremod/colony/jobs/AbstractJob.java#L65): Remove this reference to "JobLumberjack". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
1. ![MAJOR][MAJOR] [AbstractJob.java#L66](https://github.com/Minecolonies/minecolonies/blob/8a1c3c4a2518b447631ad614a066acafbb44cb24/src/main/java/com/minecolonies/coremod/colony/jobs/AbstractJob.java#L66): Remove this reference to "JobFarmer". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
1. ![MAJOR][MAJOR] [AbstractJob.java#L67](https://github.com/Minecolonies/minecolonies/blob/8a1c3c4a2518b447631ad614a066acafbb44cb24/src/main/java/com/minecolonies/coremod/colony/jobs/AbstractJob.java#L67): Remove this reference to "JobFisherman". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
1. ![MAJOR][MAJOR] [AbstractJob.java#L68](https://github.com/Minecolonies/minecolonies/blob/8a1c3c4a2518b447631ad614a066acafbb44cb24/src/main/java/com/minecolonies/coremod/colony/jobs/AbstractJob.java#L68): Remove this reference to "JobGuard". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by xavierh** [comment]
LGTM


[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1085/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Pull request merged**